src/common/micro/mode_lima_tendencies.F90

```diff
@@ -397,195 +407,207 @@ SUBROUTINE LIMA_TENDENCIES (PTSTEP, LDCOMPUTE,
 !
-   ZAI(:) =   ( XLSTT + (XCPV-XCI)*(ZT(:)-XTT) )**2 / (ZKA(:)*XRV*ZT(:)**2) &
-                + ( XRV*ZT(:) ) / (ZDV(:)*ZEISAT(:))
+   ZAI(:) =   ( CST%XLSTT + (CST%XCPV-CST%XCI)*(ZT(:)-CST%XTT) )**2 / (ZKA(:)*CST%XRV*ZT(:)**2) &
+                + ( CST%XRV*ZT(:) ) / (ZDV(:)*ZEISAT(:))
 !
-   ZCJ(:) = XSCFAC * PRHODREF(:)**0.3 / SQRT( 1.718E-5+0.0049E-5*(ZT(:)-XTT) )
+   ZCJ(:) = LIMAC%XSCFAC * PRHODREF(:)**0.3 / SQRT( 1.718E-5+0.0049E-5*(ZT(:)-CST%XTT) )
 !
 END WHERE
 !
 ! Cloud droplets : same formula for 1 and 2 moments, but using real or fixed Nc value
 ZLBDC(:)  = 1.E10
 ZLBDC3(:) = 1.E30
-WHERE (ZRCT(:)>XRTMIN(2) .AND. PCCT(:)>XCTMIN(2) .AND. LDCOMPUTE(:))
-   ZLBDC3(:) = XLBC*PCCT(:) / ZRCT(:)
-   ZLBDC(:)  = ZLBDC3(:)**XLBEXC
+ZCCT(:)=PCCT(:)
+WHERE (ZRCT(:)>LIMAP%XRTMIN(2) .AND. PCCT(:)>LIMAP%XCTMIN(2) .AND. ODCOMPUTE(:))
+   ZLBDC3(:) = LIMAW%XLBC*PCCT(:) / ZRCT(:)
+   ZLBDC(:)  = ZLBDC3(:)**LIMAW%XLBEXC
 END WHERE
 !
 ! Rain drops
 ZLBDR(:)  = 1.E10
 ZLBDR3(:) = 1.E30
-IF (NMOM_R.EQ.1) THEN
-   WHERE (ZRRT(:)>XRTMIN(3) .AND. LDCOMPUTE(:) )
-      ZLBDR(:) = XLBR*(PRHODREF(:)*ZRRT(:) )**XLBEXR
+ZCRT(:)=0.
+IF (LIMAP%NMOM_R.EQ.1) THEN
+   WHERE (ZRRT(:)>LIMAP%XRTMIN(3) .AND. ODCOMPUTE(:) )
+      ZLBDR(:) = LIMAW%XLBR*(PRHODREF(:)*ZRRT(:) )**LIMAW%XLBEXR
       ZLBDR3(:) = ZLBDR(:)**3.
+      ZCRT(:) = LIMAW%XCCR * ZLBDR(:)**LIMAW%XCXR / PRHODREF(:)
    END WHERE
-   PCRT(:) = XCCR * ZLBDR(:)**XCXR / PRHODREF(:)
 ELSE
-   WHERE (ZRRT(:)>XRTMIN(3) .AND. PCRT(:)>XCTMIN(3) .AND. LDCOMPUTE(:))
-      ZLBDR3(:) = XLBR*PCRT(:) / ZRRT(:)
-      ZLBDR(:)  = ZLBDR3(:)**XLBEXR
+   WHERE (ZRRT(:)>LIMAP%XRTMIN(3) .AND. PCRT(:)>LIMAP%XCTMIN(3) .AND. ODCOMPUTE(:))
+      ZLBDR3(:) = LIMAW%XLBR*PCRT(:) / ZRRT(:)
+      ZLBDR(:)  = ZLBDR3(:)**LIMAW%XLBEXR
+      ZCRT(:)=PCRT(:)
    END WHERE
 END IF
 !
 ! Pristine ice : same formula for 1 and 2 moments, using real or diagnosed Ni
 ZLBDI(:)  = 1.E10
-WHERE (ZRIT(:)>XRTMIN(4) .AND. PCIT(:)>XCTMIN(4) .AND. LDCOMPUTE(:))
-   ZLBDI(:) = ( XLBI*PCIT(:) / ZRIT(:) )**XLBEXI
+ZCIT(:)=PCIT(:)
+WHERE (ZRIT(:)>LIMAP%XRTMIN(4) .AND. PCIT(:)>LIMAP%XCTMIN(4) .AND. ODCOMPUTE(:))
+   ZLBDI(:) = ( LIMAC%XLBI*PCIT(:) / ZRIT(:) )**LIMAC%XLBEXI
 END WHERE
 !
 ! Snow : additional option for LSNOW_T if NMOM_S=1
 ZLBDS(:)  = 1.E10
-IF (NMOM_S.EQ.1) THEN
-   IF (LSNOW_T) THEN
-      WHERE (ZRST(:)>XRTMIN(5) .AND. LDCOMPUTE(:) )
+ZCST(:)=0.
+IF (LIMAP%NMOM_S.EQ.1) THEN
+   IF (LIMAP%LSNOW_T) THEN
+      WHERE (ZRST(:)>LIMAP%XRTMIN(5) .AND. ODCOMPUTE(:) )
          WHERE(ZT(:)>263.15)
-            ZLBDS(:) = MAX(MIN(XLBDAS_MAX, 10**(14.554-0.0423*ZT(:))),XLBDAS_MIN)
+            ZLBDS(:) = MAX(MIN(LIMAC%XLBDAS_MAX, 10**(14.554-0.0423*ZT(:))),LIMAC%XLBDAS_MIN)
          END WHERE
          WHERE(ZT(:)<=263.15)
-            ZLBDS(:) = MAX(MIN(XLBDAS_MAX, 10**(6.226-0.0106*ZT(:))),XLBDAS_MIN)
+            ZLBDS(:) = MAX(MIN(LIMAC%XLBDAS_MAX, 10**(6.226-0.0106*ZT(:))),LIMAC%XLBDAS_MIN)
          END WHERE
       END WHERE
```

Comment:
Nested where are dangerous and do not necessarily do what is expected on some compilers.